fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java

```diff
@@ -368,10 +407,11 @@ public void rename(@Nonnull final String source, @Nonnull final String dest) {
         LOGGER.trace("rename -> source={}, dest={}", source, dest);
         final byte[] key = metaSubspace.pack(source);
         CompletableFuture<byte[]> completableFuture;
-        completableFuture = context.ensureActive().get(key).exceptionally( (e) -> {
-            throw new RecordCoreArgumentException("Invalid source name in rename function for source: " + source);
-        });
-        completableFuture.thenAcceptAsync( (value) -> {
+        completableFuture = context.instrument(FDBStoreTimer.Events.LUCENE_RENAME,
+                context.ensureActive().get(key).exceptionally((e) -> {
+                    throw new RecordCoreArgumentException("Invalid source name in rename function for source: " + source);
```

Comment:
`get()` doesn't throw an exception unless there is an actual I/O problem or some other problem at the server...it just returns `null` if the value isn't there, so I don't think this exception is actually correct.

Reply:
this was actually a fun little unwrapping problem--when switching to `asyncToSync` the fact that this error wasn't being thrown (in fact an `IllegalArgumentException` was) became clear :)